corehq/apps/sms/models.py

```diff
@@ -593,6 +596,62 @@ def opt_out_sms(cls, phone_number, domain=None):
         return True
 
 
+class AbstractNumber(ABC):
+    owner_doc_type = None
+    owner_id = None
+    is_two_way = None
+    phone_number = None
+    domain = None
+
+    @property
+    @abstractmethod
+    def backend(self):
+        pass
```

Comment:
Oh, I meant to get rid of this class entirely. I had originally planned to have the `ConnectNumber` and the old `PhoneNumber` class inherit from it, and treat it as an interface the rest of the messaging code could use, but it turns out its difficult to use abstract base classes with django models because they both do weird things to the class `Meta`. I may still try to find a way to get both to pass the same type check, but I dont think this class is the answer.

Reply:
0fa683edf3acb6f103f8ee235fc710650d061c21